google-cloud-firestore/lib/google/cloud/firestore/field_path.rb

```diff
@@ -0,0 +1,211 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/firestore/convert"
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # FieldPath
+      #
+      # Represents a field path to the Firestore API. See {Client#field_path}.
+      #
+      # @example
+      #   require "google/cloud/firestore"
+      #
+      #   firestore = Google::Cloud::Firestore.new
+      #
+      #   user_snap = firestore.doc("users/frank").get
+      #
+      #   nested_field_path = Google::Cloud::Firestore::FieldPath.new(
+      #     :favorites, :food
+      #   )
+      #   user_snap.get(nested_field_path) #=> "Pizza"
+      #
+      class FieldPath
+        include Comparable
+
+        ##
+        # Creates a field path object representing the nested fields for
```

Comment:
"a nested field"

We only support one field per field path.

Reply:
Fixed.